source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -1082,18 +1082,15 @@ export class TextBlock extends BaseTextBlock {
     applyStylesTo(targetElement: HTMLElement) {
         super.applyStylesTo(targetElement);
 
-        let parentContainer = this.getParentContainer();
-        let isRtl = parentContainer ? parentContainer.isRtl() : false;
-
         switch (this.horizontalAlignment) {
             case Enums.HorizontalAlignment.Center:
                 targetElement.style.textAlign = "center";
                 break;
             case Enums.HorizontalAlignment.Right:
-                targetElement.style.textAlign = isRtl ? "left" : "right";
+                targetElement.style.textAlign = "end";
```

Comment:
That's not supported in IE... but maybe at this point we shouldn't worry anymore.

Reply:
The problem is that the root card can be unset, so I believe we'd have to compute directionality by looking at the card's ancestors all the way up to `document.dir`.